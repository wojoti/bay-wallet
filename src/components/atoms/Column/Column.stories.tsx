import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import {Column} from 'components/atoms';
import {Text, View} from 'react-native';
import {ColumnProps} from './Column';

export default {
  title: 'Atoms/Column',
  component: Column,
  decorators: [
    Story => (
      <View style={{borderWidth: 1}}>
        <Story />
      </View>
    ),
  ],
  argTypes: {
    justify: {
      control: {type: 'select'},
      options: [
        'flex-start',
        'flex-end',
        'center',
        'space-between',
        'space-around',
        'space-evenly',
        undefined,
      ],
    },
    align: {
      control: {type: 'select'},
      options: [
        'flex-start',
        'flex-end',
        'center',
        'stretch',
        'baseline',
        undefined,
      ],
    },
    background: {
      control: {type: 'color'},
    },
  },
} as ComponentMeta<typeof Column>;
const Template: ComponentStory<typeof Column> = (args: ColumnProps) => (
  <Column {...args}>
    <Text>Example text</Text>
    <Text>Example text</Text>
    <Text>Example text</Text>
  </Column>
);

export const Example = Template.bind({});
Example.args = {
  mt: 0,
  mb: 0,
  ml: 0,
  mr: 0,
  pt: 0,
  pb: 0,
  pl: 0,
  pr: 0,
  testId: '',
  align: 'flex-start',
  justify: 'flex-start',
  flex: 0,
  background: 'inherit',
};
