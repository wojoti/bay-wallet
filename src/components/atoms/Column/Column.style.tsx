import {StyleSheet} from 'react-native';

interface Props {
  mt?: number;
  mb?: number;
  ml?: number;
  mr?: number;
  pt?: number;
  pb?: number;
  pl?: number;
  pr?: number;
  justify?:
    | 'flex-start'
    | 'flex-end'
    | 'center'
    | 'space-between'
    | 'space-around'
    | 'space-evenly'
    | undefined;
  align?:
    | 'flex-start'
    | 'flex-end'
    | 'center'
    | 'stretch'
    | 'baseline'
    | undefined;
}
const style = ({mt, mb, ml, mr, pt, pb, pl, pr, justify, align}: Props) =>
  StyleSheet.create({
    wrapper: {
      marginTop: mt || 0,
      marginBottom: mb || 0,
      marginLeft: ml || 0,
      marginRight: mr || 0,
      paddingTop: pt || 0,
      paddingBottom: pb || 0,
      paddingLeft: pl || 0,
      paddingRight: pr || 0,

      justifyContent: justify || 'flex-start',
      alignItems: align || 'flex-start',
      width: '100%',
    },
  });
export default style;
