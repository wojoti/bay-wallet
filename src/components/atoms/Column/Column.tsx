import {PropsWithChildren} from 'react';
import {View} from 'react-native';
import style from './Column.style';

export type ColumnProps = PropsWithChildren<{
  mt?: number;
  mb?: number;
  ml?: number;
  mr?: number;
  pt?: number;
  pb?: number;
  pl?: number;
  pr?: number;
  justify?:
    | 'flex-start'
    | 'flex-end'
    | 'center'
    | 'space-between'
    | 'space-around'
    | 'space-evenly'
    | undefined;
  align?:
    | 'flex-start'
    | 'flex-end'
    | 'center'
    | 'stretch'
    | 'baseline'
    | undefined;
  background?: string;
  flex?: number;
  testId?: string;
}>;

const Column = ({
  mt,
  mb,
  ml,
  mr,
  pt,
  pb,
  pl,
  pr,
  justify,
  align,
  background,
  flex,
  testId,
  children,
}: ColumnProps) => {
  return (
    <View
      style={
        style({
          mt,
          mb,
          ml,
          mr,
          pt,
          pb,
          pl,
          pr,
          justify,
          align,
          background,
          flex,
        }).wrapper
      }
      testID={testId}>
      {children}
    </View>
  );
};

export default Column;
