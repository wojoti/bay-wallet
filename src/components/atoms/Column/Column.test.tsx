import '@testing-library/jest-native/extend-expect';
import {render, screen} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import Column, {ColumnProps} from './Column';

const props: ColumnProps = {
  testId: 'test-column-id',
};

const modifiedProps: ColumnProps = {
  ...props,
  mt: 14,
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<Column {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render Column', () => {
  render(<Column {...props} />);
  const ColumnElement = screen.getByTestId('test-column-id');
  expect(ColumnElement).toBeOnTheScreen();
});

test('should render column with default style', () => {
  render(<Column {...props} />);
  const columnElement = screen.getByTestId('test-column-id');
  expect(columnElement).toBeOnTheScreen();
  expect(columnElement).toHaveStyle({
    marginTop: 0,
  });
});

test('should render column with modified style through props', () => {
  render(<Column {...modifiedProps} />);
  const columnElement = screen.getByTestId('test-column-id');
  expect(columnElement).toBeOnTheScreen();
  expect(columnElement).toHaveStyle({marginTop: 14});
});
