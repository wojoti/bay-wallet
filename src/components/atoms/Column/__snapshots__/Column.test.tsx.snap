// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "flex-start",
      "backgroundColor": "inherit",
      "flex": 0,
      "justifyContent": "flex-start",
      "marginBottom": 0,
      "marginLeft": 0,
      "marginRight": 0,
      "marginTop": 0,
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 0,
    }
  }
  testID="test-column-id"
/>
`;
